File: backend/enum-mybatis-validation-app/src/main/java/com/example/enumapp/domain/order/Order.java
package com.example.enumapp.domain.order;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

public class Order {

    private Long id;
    private String customerName;
    private OrderStatus status;
    private PayMethod payMethod;
    private UserGrade userGrade;
    private LocalDateTime createdAt;
    private List<OrderItem> items = new ArrayList<>();

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getCustomerName() {
        return customerName;
    }

    public void setCustomerName(String customerName) {
        this.customerName = customerName;
    }

    public OrderStatus getStatus() {
        return status;
    }

    public void setStatus(OrderStatus status) {
        this.status = status;
    }

    public PayMethod getPayMethod() {
        return payMethod;
    }

    public void setPayMethod(PayMethod payMethod) {
        this.payMethod = payMethod;
    }

    public UserGrade getUserGrade() {
        return userGrade;
    }

    public void setUserGrade(UserGrade userGrade) {
        this.userGrade = userGrade;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public List<OrderItem> getItems() {
        return items;
    }

    public void setItems(List<OrderItem> items) {
        this.items = items;
    }
}

File: enum-jpa-querydsl-kotlin-app/src/main/kotlin/com/example/enumapp/domain/order/Order.kt
package com.example.enumapp.domain.order

import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.OneToMany
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity
@Table(name = "kqdsl_demo_orders")
open class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @Column(name = "customer_name", nullable = false, length = 100)
    open var customerName: String? = null

    @Column(name = "status", length = 20)
    open var status: OrderStatus? = null

    @Column(name = "pay_method", length = 20)
    open var payMethod: PayMethod? = null

    @Column(name = "user_grade", nullable = false, length = 20)
    open var userGrade: UserGrade? = null

    @Column(name = "created_at", nullable = false)
    open var createdAt: LocalDateTime? = null

    @OneToMany(mappedBy = "order", cascade = [CascadeType.ALL], orphanRemoval = true)
    open var items: MutableList<OrderItem> = mutableListOf()

    fun replaceItems(newItems: List<OrderItem>) {
        items.clear()
        for (item in newItems) {
            addItem(item)
        }
    }

    fun addItem(item: OrderItem) {
        items.add(item)
        item.order = this
    }
}
